camayoc/tests/qpc/cli/test_sources.py

```diff
@@ -1110,6 +1111,174 @@ def test_edit_hosts_negative(
     assert qpc_source_edit.exitstatus == 1
 
 
+@pytest.mark.parametrize('source_type, hosts, new_exclude_hosts',
+                         VALID_SOURCE_TYPE_HOSTS_WITH_EXCLUDE_HOSTS)
+def test_edit_exclude_hosts(
+        isolated_filesystem, qpc_server_config, hosts, new_exclude_hosts,
+        source_type):
+    """Edit a source's list of excluded hosts.
+
+    :id: 3ed365bc-9d5e-44ce-94c2-19dea626a138
+    :description: Use the ``qpc source edit`` command with the
+        ``--exclude-hosts`` flag to change the list of excluded hosts.
+    :steps:
+        1) Run ``qpc source add --name <name> --cred <cred>
+        --hosts <hosts> --exclude-hosts <excludedhosts> --type <type>``.
+        2) ``qpc source edit --name <name> --exclude-hosts <excludedhosts>``.
+    :expectedresults: The excluded hosts list is updated.
+    """
+    cred_name = utils.uuid4()
+    name = utils.uuid4()
+    exclude_hosts = '10.10.10.10'
+    port = default_port_for_source(source_type)
+    cred_add_and_check(
+        {
+            'name': cred_name,
+            'username': utils.uuid4(),
+            'password': None,
+            'type': source_type,
+        },
+        [(CONNECTION_PASSWORD_INPUT, utils.uuid4())],
+    )
+    qpc_source_add = pexpect.spawn(
+        """qpc source add --name {} --cred {} --hosts {} --exclude-hosts {}
+        --type {}"""
+        .format(name, cred_name, hosts, exclude_hosts, source_type)
+    )
+    assert qpc_source_add.expect('Source "{}" was added'.format(name)) == 0
+    assert qpc_source_add.expect(pexpect.EOF) == 0
+    qpc_source_add.close()
+    assert qpc_source_add.exitstatus == 0
+    source_show_and_check(
+        {'name': name},
+        generate_show_output({
+            'cred_name': cred_name,
+            'exclude_hosts': exclude_hosts,
+            'hosts': hosts,
+            'name': name,
+            'port': port,
+            'source_type': source_type,
+        })
+    )
+
+    qpc_source_edit = pexpect.spawn(
```

Comment:
You may consider creating a `source_edit_and_check` so you can use here and reuse on the next test.